integreat_cms/cms/migrations/0105_languagetreenode_languagetreenode_unique_lft_tree_and_more.py

```diff
@@ -1,8 +1,42 @@
 # Generated by Django 4.2.13 on 2024-09-25 15:01
 
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
 import django.db.models.constraints
 from django.db import migrations, models
 
+from ..utils.repair_tree import repair_tree
+
+if TYPE_CHECKING:
+    from django.apps.registry import Apps
+    from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+
+# pylint: disable=unused-argument
+
+
+def forwards_func(
```

Comment:
```suggestion
# pylint: disable=unused-argument
def forwards_func(
```

Reply:
I intentionally placed the newlines between the pylint directive and the function definition. Previously I was under the misconception that placing such a comment immediately before a function definition would disable the specified checks for only that function definition. We were very surprised to find out that instead, it disabled it for the remainder of the file *(i.e. also for `reverse_func()` and everything in `Migration` though that is already compliant)*. Leaving blank newlines best communicates that behaviour imho.
The only alternative to specifying it for the rest of the file is to put the comment at the end of the line for which the check should be disabled *(and it only affects the very line, not the statement it belongs to, which is why I didn't just put it next to the pretty printed function definition as I would then have to specify it multiple times for the same statement)*.

Reply:
Wow, that's crazy :open_mouth: 

According to the [pylint documentation](https://pylint.pycqa.org/en/latest/user_guide/messages/message_control.html#block-disables), the correct way to do it is to place the directive inside of the function, because a directive applies to its containing block...

I guess we should create a issue to fix all of our pylint directives then? :laughing:

Reply:
Oooh, so to only apply to the next line you need `# pylint: disable-next=…` instead of `# pylint: disable=…` – this might just have started as an oversight then

Reply:
Okay, this is interesting. I prefer to have `reverse_func()` in for completeness sake, but just using
```python
def reverse_func(
    apps: Apps,
    schema_editor: BaseDatabaseSchemaEditor,
) -> None:
    """
    Data migration when reverting this migration
    (nothing to do since the data is already consistent)
    """
    # pylint: disable=unused-argument
```
apparently *doesn't* count the disable towards the block – I get a warning for each of the unused arguments, and then one for the useless suppression of `useless-argument`.
If I only add a `pass` to the end, the comment *does* count towards the block, and pylint only explains to me that I really don't need the `pass` statement, since the docstring is a multiline string that already counts as a statement, making the function not empty already. :grin: